core/src/com/unciv/logic/GameStarter.kt

```diff
@@ -167,9 +168,23 @@ object GameStarter {
         availableCityStatesNames.addAll(ruleset.nations.filter { it.value.isCityState() }.keys
                 .shuffled().sortedByDescending { it in cityStatesWithStartingLocations })
 
+        val unusedMercantileResources = ruleset.tileResources.values.filter { it.unique == "Can only be created by Mercantile City-States" }.toMutableList()
+
         for (cityStateName in availableCityStatesNames.take(newGameParameters.numberOfCityStates)) {
             val civ = CivilizationInfo(cityStateName)
             civ.cityStatePersonality = CityStatePersonality.values().random()
+            if (ruleset.nations[cityStateName]?.cityStateType == CityStateType.Mercantile) {
+                if (!ruleset.tileResources.values.any { it.unique == "Can only be created by Mercantile City-States" }) {
```

Comment:
Maybe cache `unusedMercantileResources.isEmpty()` outside the loop? Two uses inside the loop...

Reply:
Sorry, I'm not sure what you mean by this? Do you think I'm using an inappropriate data structure here? I wasn't bothered about optimizing this bit too heavily as it only gets called once.

Reply:
I just saw like an obvious small optimization that would also improve readability.. But you're right nothing critical here. And I was wrong as well as the second reuse is not the bool 'existence' but the complete list.